File: 1435-xor-queries-of-a-subarray/xor-queries-of-a-subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = arr.size();
        vector<int> Xor(n);
        Xor[0] = arr[0];
        for(int i = 1 ; i < n ; i++){
            Xor[i] = Xor[i-1]^arr[i]; 
        }
        vector<int> ans;
        for(auto it : queries){
            int first = it[0];
            int last = it[1];
            if ( first == 0 ){
                ans.push_back(Xor[last]);
            }
            else ans.push_back(Xor[last]^Xor[first-1]);
        }
        return ans;
    }
};

File: 1435-xor-queries-of-a-subarray/xor-queries-of-a-subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        N = len(arr)
        pre = [0]*N
        pre[0] = arr[0]

        for i in range(1,N):
            pre[i] = pre[i-1]^arr[i]

        sz = len(queries)
        print(pre)
        ans = [0]*sz
        for i,(l,r) in enumerate(queries):
            if l > 0:
                ans[i] = pre[r]^pre[l-1]
            else:
                ans[i] = pre[r]
        
        return ans
        